decidim/decidim
# Max characters for questionnaire answers

#### :tophat: What? Why?
Allow for questionnaire questions to have a character limit option to ensure user answers to questions that allow free text are not too long.

It adds a new integer attribute to `Decidim::Forms::Question`, and it uses `InputCharacterCounter` to manage client-side logic.

#### :pushpin: Related Issues
- Related to [Proposal in MetaDecidim](https://meta.decidim.org/processes/roadmap/f/122/proposals/16222)

#### Testing
- Admin ➡️  Manage a survey
- Add a question (any type, if it has multiple options check "free text" for one of them)
- Visit survey public page
- Answer survey
- Check that the text field has a character limit
- Check that a message "X characters remaining" appears below the text field

#### :clipboard: Checklist

### :camera: Screenshots

![Answer survey with character limit](https://user-images.githubusercontent.com/8806781/107240064-c9fdf800-6a29-11eb-9259-8cbb674aeeb3.png)


![Set max characters for question](https://user-images.githubusercontent.com/8806781/107239678-65db3400-6a29-11eb-9e0d-bc8f1a177844.png)


:hearts: Thank you!
